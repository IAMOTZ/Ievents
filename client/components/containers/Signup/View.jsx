import React from 'react';
import PropTypes from 'prop-types';
import { RegularTopNavigation } from '../../common/TopNavigation';
import Footer from '../../common/Footer';
import { LoadingIcon } from '../../common/LoadingAnimation';
import { BigAlert, SmallAlert } from '../../common/Alert';

const View = props => (
  <div id="sign-up-container">
    <RegularTopNavigation />
    <div className="d-flex flex-column align-items-center page-content">
      <LoadingIcon start={props.loggingUserStarted} size={3} />
      <div className="card card-form">
        <h1 className="card-header">Sign up</h1>
        <div className="card-body">
          <BigAlert message={props.loggingUserError} />
          <form>
            <div className="form-group">
              <label htmlFor="name">Name</label>
              <div className="input-group">
                <div className="input-group-addon">
                  <i className="fa fa-user" />
                </div>
                <input
                  id="name"
                  type="text"
                  name="name"
                  className="form-control"
                  placeholder="Name"
                  onChange={props.getInput}
                />
              </div>
              <small
                className="form-text text-muted"
              >Can contain letters and numbers alone
              </small>
              <SmallAlert message={props.inputErrors.nameError} />
            </div>
            <div className="form-group">
              <label htmlFor="email">Email</label>
              <div className="input-group">
                <div className="input-group-addon">
                  <i className="fa fa-envelope" />
                </div>
                <input
                  id="email"
                  type="email"
                  name="email"
                  className="form-control"
                  placeholder="Your Email"
                  onChange={props.getInput}
                />
              </div>
              <SmallAlert message={props.inputErrors.emailError} />
            </div>
            <div className="form-group">
              <label htmlFor="password">Password</label>
              <div className="input-group">
                <div className="input-group-addon">
                  <i className="fa fa-user-secret" />
                </div>
                <input
                  id="password"
                  name="password"
                  type="password"
                  className="form-control"
                  placeholder="Password"
                  onChange={props.getInput}
                />
              </div>
              <small
                className="form-text text-muted"
              >Must contain capital, small letters and numbers
              </small>
              <SmallAlert message={props.inputErrors.passwordError} />
            </div>
            <div className="form-group">
              <label htmlFor="password">Confirm Password</label>
              <div className="input-group">
                <div className="input-group-addon">
                  <i className="fa fa-user-secret" />
                </div>
                <input
                  id="confirm-password"
                  name="confirmPassword"
                  type="password"
                  className="form-control"
                  placeholder="Confirm Password"
                  onChange={props.getInput}
                />
              </div>
              <SmallAlert message={props.inputErrors.confirmPasswordError} />
            </div>
            <button
              className="btn btn-block ie-blue-button mt-4"
              disabled={props.loggingUserStarted}
              onClick={props.register}
            >Register
            </button>
          </form>
        </div>
      </div>
    </div>
    <Footer />
  </div>
);

View.defaultProps = {
  loggingUserError: '',
  inputErrors: {},
};

/* eslint-disable react/forbid-prop-types */
View.propTypes = {
  loggingUserStarted: PropTypes.bool.isRequired,
  loggingUserError: PropTypes.string,
  inputErrors: PropTypes.object,
  getInput: PropTypes.func.isRequired,
  register: PropTypes.func.isRequired,
};

export default View;
