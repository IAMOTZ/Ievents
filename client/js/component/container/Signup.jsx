import React from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import TopNavigation from '../common/TopNavigation.jsx';
import Footer from '../common/Footer.jsx';
import { WarningAlert } from '../common/Alert';

import { createUser, clearError } from '../../actions/authAction';

@connect((store) => {
  return {
    user: store.user,
    error: store.user.status.error.message
  }
})

export default class Signup extends React.Component {
  constructor() {
    super();
    this.state = {
      name: null,
      email: null,
      password: null,
      confirmPassword: null,
    };
  }

  componentWillUnmount() {
    this.props.dispatch(clearError());
  }

  // This method uses user input to update the state
  getInput = (e) => {
    const state = this.state;
    state[e.target.name] = e.target.value;
    this.setState(state);
  }

  // This method fires the action to create a user
  register = () => {
    const {
      name, email, password, confirmPassword,
    } = this.state;
    this.props.dispatch(createUser({
      name,
      email,
      password,
      confirmPassword,
    }));
  }

  render() {
    if (this.props.user.status.fetched === true) {
      return (<Redirect to="/addEvent" />);
    } else {
      return (
        <div id="sign-up-container">
          <TopNavigation />
          <div className="d-flex flex-column align-items-center">
            <div className="card card-form">
              <WarningAlert message={this.props.error}/>
              <h1 className="card-header">Sign up</h1>
              <div className="card-body">
                <form>
                  <div className="form-group">
                    <label htmlFor="name">Name</label>
                    <div className="input-group">
                      <div className="input-group-addon">
                        <i className="fa fa-user" />
                      </div>
                      <input type="text" className="form-control" id="name" placeholder="Name" name="name" onChange={this.getInput} />
                    </div>
                  </div>
                  <div className="form-group">
                    <label htmlFor="email">Email</label>
                    <div className="input-group">
                      <div className="input-group-addon">
                        <i className="fa fa-envelope" />
                      </div>
                      <input type="email" className="form-control" id="email" name="email" placeholder="Your Email" onChange={this.getInput} />
                    </div>
                  </div>
                  <div className="form-group">
                    <label htmlFor="password">Password</label>
                    <div className="input-group">
                      <div className="input-group-addon">
                        <i className="fa fa-user-secret" />
                      </div>
                      <input type="password" className="form-control" id="password" name="password" placeholder="Password" onChange={this.getInput} />
                    </div>
                  </div>
                  <div className="form-group">
                    <label htmlFor="password">Confirm Password</label>
                    <div className="input-group">
                      <div className="input-group-addon">
                        <i className="fa fa-user-secret" />
                      </div>
                      <input type="password" className="form-control" id="password" name="confirmPassword" placeholder="Confirm Password" onChange={this.getInput} />
                    </div>
                  </div>
                  <a className="btn btn-block dark-button text-white" onClick={this.register} >Register</a>
                </form>
              </div>
            </div>

            <Footer />
          </div>
        </div>

      );
    }

  }
}
