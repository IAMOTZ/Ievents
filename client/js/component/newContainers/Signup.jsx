import React from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { validateSignupInputs } from '../../helpers/inputValidators';
import { createUser } from '../../actions/authAction';
import { stopAsyncProcess } from '../../actions/commonActions';
import * as asyncProcess from '../../actions/asyncProcess';
import TopNavigation from '../common/TopNavigation';
import Footer from '../common/Footer';
import { LoadingIcon } from '../common/LoadingAnimation';
import { BigAlert, SmallAlert } from '../common/Alert';

@connect(store => (
  {
    user: store.authReducer.user,
    loggingUserStarted: store.authReducer.loggingUserStarted,
    loggingUserResolved: store.authReducer.loggingUserResolved,
    loggingUserError: store.authReducer.loggingUserResolved,
  }
))
class Signup extends React.Component {
  constructor() {
    super();
    this.state = {
      name: null,
      email: null,
      password: null,
      confirmPassword: null,
      inputErrors: {
        nameError: null,
        emailError: null,
        passwordError: null,
        confirmPasswordError: null,
      },
    };
  }

  componentWillUnmount() {
    this.props.dispatch(stopAsyncProcess(asyncProcess.ADDING_USER));
  }

  /**
   * Stores the user inputs in the state of this component.
   * @param {Event} event The event object.
   */
  getInput = (event) => {
    const state = { ...this.state };
    state[event.target.name] = event.target.value;
    this.setState(state);
  }

  /**
   * Clears all the inputErrors in the state.
   */
  clearInputErrors = () => {
    const state = { ...this.state };
    state.inputErrors = {
      nameError: null,
      emailError: null,
      passwordError: null,
      confirmPasswordError: null,
    };
    this.setState(state);
  }

  /**
   * It Dispatches an action to register a user.
   * @param {Event} event The event object.
   */
  register = (event) => {
    event.preventDefault();
    this.props.dispatch(stopAsyncProcess(asyncProcess.ADDING_USER));
    const {
      name, email, password, confirmPassword,
    } = this.state;
    const inputErrors = validateSignupInputs(this.state);
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      this.clearInputErrors();
      this.props.dispatch(createUser({
        name, email, password, confirmPassword,
      }));
    }
  }

  render() {
    if (this.props.loggingUserResolved) {
      return (<Redirect to="/addEvent" />);
    }
    return (
      <div id="sign-up-container">
        <TopNavigation />
        <div className="d-flex flex-column align-items-center main-content">
          <LoadingIcon start={this.props.loggingUserStarted} size={3} />
          <div className="card card-form">
            <h1 className="card-header">Sign up</h1>
            <div className="card-body">
              <BigAlert message={this.props.loggingUserError} />
              <form>
                <div className="form-group">
                  <label htmlFor="name">Name</label>
                  <div className="input-group">
                    <div className="input-group-addon">
                      <i className="fa fa-user" />
                    </div>
                    <input
                      id="name"
                      type="text"
                      name="name"
                      className="form-control"
                      placeholder="Name"
                      onChange={this.getInput}
                    />
                  </div>
                  <small
                    className="form-text text-muted"
                  >Can contain letters and numbers alone
                    </small>
                  <SmallAlert message={this.state.inputErrors.nameError} />
                </div>
                <div className="form-group">
                  <label htmlFor="email">Email</label>
                  <div className="input-group">
                    <div className="input-group-addon">
                      <i className="fa fa-envelope" />
                    </div>
                    <input
                      id="email"
                      type="email"
                      name="email"
                      className="form-control"
                      placeholder="Your Email"
                      onChange={this.getInput}
                    />
                  </div>
                  <SmallAlert message={this.state.inputErrors.emailError} />
                </div>
                <div className="form-group">
                  <label htmlFor="password">Password</label>
                  <div className="input-group">
                    <div className="input-group-addon">
                      <i className="fa fa-user-secret" />
                    </div>
                    <input
                      id="password"
                      name="password"
                      type="password"
                      className="form-control"
                      placeholder="Password"
                      onChange={this.getInput}
                    />
                  </div>
                  <small
                    className="form-text text-muted"
                  >Must contain capital, small letters and numbers
                    </small>
                  <SmallAlert message={this.state.inputErrors.passwordError} />
                </div>
                <div className="form-group">
                  <label htmlFor="password">Confirm Password</label>
                  <div className="input-group">
                    <div className="input-group-addon">
                      <i className="fa fa-user-secret" />
                    </div>
                    <input
                      id="confirm-password"
                      name="confirmPassword"
                      type="password"
                      className="form-control"
                      placeholder="Confirm Password"
                      onChange={this.getInput}
                    />
                  </div>
                  <SmallAlert message={this.state.inputErrors.confirmPasswordError} />
                </div>
                <button
                  className="btn btn-block dark-button text-white mt-4"
                  disabled={this.props.loggingUserStarted}
                  onClick={this.register}
                >Register
                  </button>
              </form>
            </div>
          </div>
          <Footer />
        </div>
      </div>
    );
  }
}

export default Signup;
