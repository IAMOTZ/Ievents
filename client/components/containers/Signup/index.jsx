import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { validateSignupInputs } from '../../../helpers/inputValidators';
import { createUser } from '../../../actions/authAction';
import { stopAsyncProcess } from '../../../actions/commonActions';
import * as asyncProcess from '../../../actions/asyncProcess';
import './styles.scss';
import View from './View';

@connect(store => (
  {
    loggingUserStarted: store.authReducer.loggingUserStarted,
    loggingUserResolved: store.authReducer.loggingUserResolved,
    loggingUserError: store.authReducer.loggingUserError,
  }
))
class Signup extends React.Component {
  constructor() {
    super();
    this.state = {
      name: null,
      email: null,
      password: null,
      confirmPassword: null,
      inputErrors: {
        nameError: null,
        emailError: null,
        passwordError: null,
        confirmPasswordError: null,
      },
    };
  }

  componentWillUnmount() {
    this.props.dispatch(stopAsyncProcess(asyncProcess.ADDING_USER));
  }

  /**
   * Stores the user inputs in the state of this component.
   * @param {Event} event The event object.
   */
  getInput = (event) => {
    const state = { ...this.state };
    state[event.target.name] = event.target.value;
    this.setState(state);
  }

  /**
   * Clears all the inputErrors in the state.
   */
  clearInputErrors = () => {
    const state = { ...this.state };
    state.inputErrors = {
      nameError: null,
      emailError: null,
      passwordError: null,
      confirmPasswordError: null,
    };
    this.setState(state);
  }

  /**
   * It Dispatches an action to register a user.
   * @param {Event} event The event object.
   */
  register = (event) => {
    event.preventDefault();
    this.props.dispatch(stopAsyncProcess(asyncProcess.ADDING_USER));
    const {
      name, email, password, confirmPassword,
    } = this.state;
    const inputErrors = validateSignupInputs(this.state);
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      this.clearInputErrors();
      this.props.dispatch(createUser({
        name, email, password, confirmPassword,
      }));
    }
  }

  render() {
    if (this.props.loggingUserResolved) {
      return (<Redirect to="/addEvent" />);
    }
    return (
      <View
        loggingUserStarted={this.props.loggingUserStarted}
        loggingUserError={this.props.loggingUserError}
        inputErrors={this.state.inputErrors}
        getInput={this.getInput}
        register={this.register}
      />);
  }
}

Signup.defaultProps = {
  loggingUserStarted: false,
  loggingUserResolved: false,
  loggingUserError: '',
  dispatch: () => {},
};

Signup.propTypes = {
  loggingUserStarted: PropTypes.bool,
  loggingUserResolved: PropTypes.bool,
  loggingUserError: PropTypes.string,
  dispatch: PropTypes.func,
};

export default Signup;
