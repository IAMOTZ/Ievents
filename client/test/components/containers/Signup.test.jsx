import React from 'react';
import Signup from '../../../components/containers/Signup';
import SignupView from '../../../components/containers/Signup/View';

const mockEvent = target => ({
  preventDefault: jest.fn(), target,
});

const sampleUserData = {
  name: 'tester',
  email: 'tester@gmail.com',
  password: 'Password123',
  confirmPassword: 'Password123'
};

describe('<Signup />', () => {
  const props = {
    dispatch: jest.fn(),
    loggingUserStarted: false,
    loggingUserResolved: false,
    loggingUserError: null,
  };
  const alterProps = newProps => ({ ...props, ...newProps });
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Signup {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should redirect to centers page if logging resolved', () => {
      wrapper.setProps(alterProps({ loggingUserResolved: true }));
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Behaviour:', () => {
    let wrapper; let wrapperInstance;
    beforeEach(() => {
      props.dispatch.mockReset();
      wrapper = shallow(<Signup {...props} />);
      wrapperInstance = wrapper.instance();
    });
    // The names on the following test block refers
    // to the instance method that handles the behaviour tested.
    describe('GET_INPUT', () => {
      it('should update the state with user input', () => {
        const userInput = 'tester';
        wrapperInstance.getInput(mockEvent({ name: 'name', value: userInput }));
        expect(wrapper.state('name')).toEqual(userInput);
      });
    });
    describe('CLEAR_INPUT_ERRORS', () => {
      it('should clear input errors in the state', () => {
        const nameError = 'You name has errors';
        wrapper.setState({ inputErrors: { nameError } });
        wrapperInstance.clearInputErrors();
        expect(wrapper.state('inputErrors').nameError).toBeNull();
      });
    });

    describe('REGISTER', () => {
      it('should dispatch an action and set inputErrors if validation fails', () => {
        const wrongEmail = 'testergmail.com';
        wrapper.setState({ email: wrongEmail });
        wrapperInstance.register(mockEvent());
        expect(props.dispatch.mock.calls.length).toEqual(1);
        expect(typeof wrapper.state('inputErrors').emailError).toEqual('string');
      });
      it('should clear inputErrors and dispatch 2 actions if validation pass', () => {
        const previousError = 'Email is required';
        wrapper.setState({
          ...sampleUserData, inputErrors: { emailError: previousError }
        });
        wrapperInstance.register(mockEvent());
        expect(wrapper.state('inputErrors').emailError).toBeNull();
        expect(props.dispatch.mock.calls.length).toEqual(2);
      });
    });

    describe('COMOPONENT_WILL_UNMOUNT', () => {
      it('should dispatch an action when the component wants to unmount', () => {
        wrapper.unmount();
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });
  });
});

describe('<SignupView />', () => {
  const props = {
    getInput: jest.fn(),
    inputErrors: {},
    loggingUserError: null,
    loggingUserStarted: false,
    register: jest.fn(),
  };
  describe('Rendering', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<SignupView {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
