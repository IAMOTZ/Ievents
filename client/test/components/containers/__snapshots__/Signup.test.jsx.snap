// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> Rendering: should redirect to centers page if logging resolved 1`] = `
<Redirect
  push={false}
  to="/centers"
/>
`;

exports[`<Signup /> Rendering: should render correctly 1`] = `
<View
  getInput={[Function]}
  inputErrors={
    Object {
      "confirmPasswordError": null,
      "emailError": null,
      "nameError": null,
      "passwordError": null,
    }
  }
  loggingUserError={null}
  loggingUserStarted={false}
  register={[Function]}
/>
`;

exports[`<SignupView /> Rendering should render correctly 1`] = `
<div
  id="sign-up-container"
>
  <RegularTopNavigation />
  <div
    className="d-flex flex-column align-items-center page-content"
  >
    <LoadingIcon
      size={3}
      start={false}
    />
    <div
      className="card card-form"
    >
      <h1
        className="card-header"
      >
        Sign up
      </h1>
      <div
        className="card-body"
      >
        <BigAlert
          message={null}
        />
        <form>
          <div
            className="form-group"
          >
            <label
              htmlFor="name"
            >
              Name
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-addon"
              >
                <i
                  className="fa fa-user"
                />
              </div>
              <input
                className="form-control"
                id="name"
                name="name"
                onChange={[MockFunction]}
                placeholder="Name"
                type="text"
              />
            </div>
            <SmallAlert />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-addon"
              >
                <i
                  className="fa fa-envelope"
                />
              </div>
              <input
                className="form-control"
                id="email"
                name="email"
                onChange={[MockFunction]}
                placeholder="Your Email"
                type="email"
              />
            </div>
            <SmallAlert />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-addon"
              >
                <i
                  className="fa fa-user-secret"
                />
              </div>
              <input
                className="form-control"
                id="password"
                name="password"
                onChange={[MockFunction]}
                placeholder="Password"
                type="password"
              />
            </div>
            <small
              className="form-text text-muted"
            >
              Must contain capital, small letters and numbers
            </small>
            <SmallAlert />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="password"
            >
              Confirm Password
            </label>
            <div
              className="input-group"
            >
              <div
                className="input-group-addon"
              >
                <i
                  className="fa fa-user-secret"
                />
              </div>
              <input
                className="form-control"
                id="confirm-password"
                name="confirmPassword"
                onChange={[MockFunction]}
                placeholder="Confirm Password"
                type="password"
              />
            </div>
            <SmallAlert />
          </div>
          <button
            className="btn btn-block ie-blue-button mt-4"
            disabled={false}
            onClick={[MockFunction]}
          >
            Register
          </button>
          <div
            className="text-center mt-2"
            style={
              Object {
                "fontSize": "15px",
              }
            }
          >
            <span>
              Already registered?
            </span>
            Â 
            <Link
              replace={false}
              to="/users/login"
            >
              Log in
            </Link>
          </div>
        </form>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;
